app/bundles/ReportBundle/Event/ReportGeneratorEvent.php

```diff
@@ -374,14 +374,16 @@ public function applyTagFilter(CompositeExpression $groupExpr, array $filter): v
         }
 
         if (in_array($filter['condition'], ['in', 'notEmpty'])) {
-            $groupExpr->add(
+            $groupExpr = $groupExpr->with(
                 $tagSubQuery->expr()->in('l.id', $tagSubQuery->getSQL())
             );
         } elseif (in_array($filter['condition'], ['notIn', 'empty'])) {
-            $groupExpr->add(
+            $groupExpr = $groupExpr->with(
                 $tagSubQuery->expr()->notIn('l.id', $tagSubQuery->getSQL())
             );
         }
+
+        return $groupExpr;
```

Comment:
I'm wondering how come this change which makes this method immutable did not break the `testApplyTagFilter` test which is using it as a mutable method 🤷

Reply:
Aha, it was broken and fixed in the meantime :) https://github.com/mautic/mautic/pull/12318/commits/8038557555dde8cb2acec7a03948e73e564e92fb

Reply:
Here is the change [in the test](https://github.com/mautic/mautic/pull/12318/files#diff-2130e60721761f2c6ad1d83feb2420e2e2f6207777e23833967675b4623fac79R322)